setup_build.py

```diff
@@ -35,20 +35,22 @@ def localpath(*args):
               localpath("lzf/lzf/lzf_c.c"),
               localpath("lzf/lzf/lzf_d.c")]}
 
+COMPILER_SETTINGS = {
+   'libraries'      : ['hdf5', 'hdf5_hl'],
+   'include_dirs'   : [localpath('lzf')],
+   'library_dirs'   : [],
+   'define_macros'  : [('H5_USE_16_API', None)]
+}
 
 if sys.platform.startswith('win'):
-    COMPILER_SETTINGS = {
-        'libraries'     : ['h5py_hdf5', 'h5py_hdf5_hl'],
-        'include_dirs'  : [localpath('lzf'), localpath('windows')],
-        'library_dirs'  : [],
-        'define_macros' : [('H5_USE_16_API', None), ('_HDF5USEDLL_', None)] }
-
+    COMPILER_SETTINGS['include_dirs'].append(localpath('windows'))
+    COMPILER_SETTINGS['define_macros'].extend([
+        ('_HDF5USEDLL_', None),
+        ('H5_BUILT_AS_DYNAMIC_LIB', None)
+    ])
 else:
-    COMPILER_SETTINGS = {
-       'libraries'      : ['hdf5', 'hdf5_hl'],
-       'include_dirs'   : [localpath('lzf'), '/opt/local/include', '/usr/local/include'],
-       'library_dirs'   : ['/opt/local/lib', '/usr/local/lib'],
-       'define_macros'  : [('H5_USE_16_API', None)] }
+    COMPILER_SETTINGS['include_dirs'].extend(['/opt/local/include''/usr/local/include'])
+    COMPILER_SETTINGS['library_dirs'].extend(['/opt/local/include''/usr/local/include'])
```

Comment:
These needs commas between them. Otherwise the strings are concatenated together.

Reply:
NVM. Was fixed in PR ( https://github.com/h5py/h5py/pull/883 ).